### DISCUSSION TITLE
New here

### DISCUSSION THREAD

**Discussion by skempohlee** [community/community#120050] [Programming Help]
### Body

Please can anyone help me out with my code im new to programming 
Just trying to find my way around 

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ebndev** [comment]
Hi @skempohlee, we're happy you're here 😀

I encourage you to post your specific questions to the category, as we do now allow asks for recruitment on helping with projects / assignments.

You are more likely to get a useful response if you are explicit about what your project entails, giving a few more details might help someone give you a nudge in the right direction.

Best of luck!

**Comment by RM25BAM** [comment]
Hi @skempohlee, I'm glad you are starting to learn to code. That's a great thing, mostly when you are trying to focus on version control, very important tool. Plus GitHub community is really helpful. There is a lot of resources online that can also help like YouTube, stack overflow, and google. Don't give up and keep going !!!! :-)

**Reply by ebndev** [reply]
@RM25BAM Thanks for encouraging new users! 🥳

**Comment by skempohlee** [comment]
Thanks,actually I’ve been finding this assignment,it just keeps bringing “error syntax “difficult and I’ve submitted it the way I understand it,so im going to post it here.

This is the assignment.

Task 1: Build a function-based console log message generator
In this exercise, your task is to code a function named consoleStyler, which accepts four parameters:

color
background
fontSize
txt
Inside the body of the consoleStyler() function declaration, you need to do the following:

Create a new variable named message, and assign the following to it on the very first line inside the consoleStyler() function body.:

"%c" + txt;
Tip: Do not copy the 3 back ticks. These are used to format this document in the Preview tab.

Create a style variable and assign the following to it on the next line:

`color: ${color};`
Tip: The single backtick before color and after the semi-colon must be included.

Next, update the style variable (using the += operator) with the following code:

`background: ${background};`
Tip: The single backtick before background and after the semi-colon must be included.

Then, update the style variable (again, using the += operator) with the following code:

`font-size: ${fontSize};`
Tip: The single backtick before font-size and after the semi-colon must be included.

Finally, console log the message and style variables inside the consoleStyler function declaration.

Hint: Be sure to use backticks (``) when updating your variable styles and not single ('') or double ("") quotes.


Task 2: Build another console log message generator.
Your task is to code another function, and name it celebrateStyler(). The function accepts a single parameter, reason, which should be of string data type.

Inside the function declaration's body, code the following:

A new variable, named fontStyle, assigning it this code:

"color: tomato; font-size: 50px";
On the next line, an if statement, verifying that reason == "birthday".

Inside the body of the if block, code the following:

console.log(`%cHappy birthday`, fontStyle);
On the next line, add an else if, and inside the parentheses, check that

reason == "champions"
Inside the else if block, add this code:

console.log(`%cCongrats on the title!`, fontStyle);
Add an else block, with the following code inside of it:

console.log(message, style);

Task 3: Run both the consoleStyler and the celebrateStyler functions
Invoke the consoleStyler() function, with the following arguments:

'#1d5c63'

'#ede6db'

'40px'

'Congrats!'

Next, invoke the celebrateStyler() function, with the following argument:

'birthday'
Task 4: Insert a congratulatory and custom message
Code another function, named styleAndCelebrate().
The function declaration's body should consist of two function invocations:

consoleStyler(color, background, fontSize, txt);  
celebrateStyler(reason);
Next, invoke the new function, using the following arguments:

'ef7c8e'
'fae8e0'
'30px'
'You made it!'
'champions'
![F5C5AAE8-2FEB-437F-8DB4-0AD79B06DD0A](https://github.com/community/community/assets/155685051/d9d39547-2895-405d-aa17-9c709ced66e7)





My code that i submitted 

function consoleStyler(color, background, fontSize, txt) {
  let message = "%c" + txt;
  let style = `color: ${color}; background: ${background}; font-size: ${fontSize};`;
  console.log(message, style);
}

function celebrateStyler(reason) {
  let fontStyle = "color: tomato; font-size: 50px;"; // Added semicolon at the end
  if (reason === "birthday") {
    console.log(`%cHappy birthday`, fontStyle);
  } else if (reason === "champions") {
    console.log(`%cCongrats on the title!`, fontStyle);
  } else {
    console.log("Unknown reason:", reason);
  }
}

consoleStyler('#1d5c63', '#ede6db', '40px', 'Congrats!');
celebrateStyler('birthday');

function styleAndCelebrate(color, background, fontSize, txt, reason) {
  consoleStyler(color, background, fontSize, txt);
  celebrateStyler(reason);
}

styleAndCelebrate('ef7c8e', 'fae8e0', '30px', 'You made it!', 'champions');

**Reply by ebndev** [reply]
From the code submission:

`let style = color: ${color}; background: ${background}; font-size: ${fontSize};;` this is missing the backticks that would create the string. That might've been the syntax error the console was referring to.

Side note: when you pass in arguments, just make sure to add # in front of the hex codes for color values if you were to use this code later on to make styling changes :) @ `styleAndCelebrate('ef7c8e', 'fae8e0', '30px', 'You made it!', 'champions');`

**Reply by skempohlee** [reply]
Hi @ebndev Thanks,I really appreciate your comments it was helpful,im on this journey to make changes and I must to.
But something isn’t clear to me 
I want to start practicing writing my own code on VScode but I don’t understand how to  launch so it’ll be a website,I’ve studied and completed HTML and CSS so I wanna put them into practice

**Reply by ebndev** [reply]
I'm glad it was helpful! If it answered your question, feel free to "Mark as answer" so users know there's an answer to this discussion.

To launch your code, you can download an extension in VS Code called "Open in Browser", then right click in your HTML file to "open in browser" and you can see your site as you're working on it.

As for hosting your HTML & CSS website for it to be online, you can use GitHub Pages 😁 Here's a tutorial on [Creating a GitHub Pages site](https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site) and if you have any questions, I highly encourage you to post in our [Pages](https://github.com/orgs/community/discussions/categories/pages) category. 

For more VS Code / coding questions in general, you can continue to create new discussions in this category (Programming Help) as well. 

Best of luck!

**Comment by mainajackson95** [comment]
Welcome to the platform

**Comment by mainajackson95** [comment]
How are you fairing here
